src/nvim/event/process.c

```diff
@@ -21,9 +21,12 @@
 # include "event/process.c.generated.h"
 #endif
 
-// Time (ns) for a process to exit cleanly before we send TERM/KILL.
-#define TERM_TIMEOUT 1000000000
-#define KILL_TIMEOUT (TERM_TIMEOUT * 2)
+// Time for a process to exit cleanly before we send KILL.
+#define KILL_TIMEOUT_MS 2000
+#define KILL_TIMEOUT_NS (KILL_TIMEOUT_MS * 1000000)
+// Time for a pty process to exit due to SIGHUP before we send TERM.
+#define PTY_TERM_TIMEOUT_MS 1000
+#define PTY_TERM_TIMEOUT_NS (PTY_TERM_TIMEOUT_MS * 1000000)
```

Comment:
2s vs 1s doesn't really matter, at least I can't think of a reason why it should be different for pty vs non-pty jobs.